src/components/LoginForm/LoginForm.tsx

```diff
@@ -20,26 +23,41 @@ const formSchema = z.object({
 	password: z.string().min(8, {
 		message: 'Password must be at least 8 characters long.',
 	}),
-});
+})
 
 export function LoginForm() {
+	const [isLoading, setIsLoading] = useState(false)
+	const [error, setError] = useState<string | null>(null)
+	const navigate = useNavigate()
+
 	// 1. Define your form.
 	const form = useForm<z.infer<typeof formSchema>>({
 		resolver: zodResolver(formSchema),
 		defaultValues: {
 			email: '',
 			password: '',
 		},
-	});
+	})
 
 	// 2. Define a submit handler.
-	function onSubmit(values: z.infer<typeof formSchema>) {
-		console.log(values);
+	async function onSubmit(values: z.infer<typeof formSchema>) {
+		setIsLoading(true)
+		setError(null)
+		try {
+			await signIn(values.email, values.password)
+			navigate('/dashboard')
+		} catch (error) {
+			setError('Failed to sign in. Please check your credentials.')
+			console.error('Login error:', error)
+		} finally {
+			setIsLoading(false)
+		}
 	}
 
 	return (
 		<Form {...form}>
 			<form onSubmit={form.handleSubmit(onSubmit)} className='space-y-8'>
+				{error && <div className='text-red-500 text-center'>{error}</div>}
```

Comment:
please use these classNames for styling: `text-sm font-medium text-destructive text-center` for consistency with other forms.

Reply:
Thanks for the reminder